contracts/FeralfileArtworkV4_2.sol

```diff
@@ -0,0 +1,184 @@
+            : require(
+                saleData_.price == msg.value,
+                "FeralfileExhibitionV4: invalid payment amount"
+            );
+
+        bytes32 message = keccak256(
+            abi.encode(block.chainid, address(this), saleData_)
+        );
+
+        //check nonce
+        _useCheckedNonce(saleData_.destination, saleData_.nonce);
+
+        if (!isValidSignature(message, r_, s_, v_)) {
+            revert InvalidSignature();
+        }
+
+        uint256 itemRevenue;
+        if (saleData_.price > saleData_.cost) {
+            itemRevenue =
+                (saleData_.price - saleData_.cost) /
+                saleData_.quantity;
+        }
+
+        uint256 nextSaleTokenId = seriesNextSaleTokenIds[saleData_.seriesID];
+        uint256 i = 0;
+        while (i < saleData_.quantity) {
+            uint256 tokenIdForSale = nextSaleTokenId;
+
+            if (!_exists(tokenIdForSale)) {
+                revert TokenIDNotFound();
+            }
+
+            nextSaleTokenId++;
+            if (ownerOf(tokenIdForSale) != address(this)) {
+                continue;
+            }
+
+            // send NFT
+            _safeTransfer(
+                address(this),
+                saleData_.destination,
+                tokenIdForSale,
+                ""
+            );
+
+            emit BuyArtworkV2(saleData_.destination, tokenIdForSale, saleData_.nonce);
+            i++;
+        }
+
+        // save next sale token id for seriesID
+        seriesNextSaleTokenIds[saleData_.seriesID] = nextSaleTokenId;
+
+        // distribute royalty
+        uint256 distributedRevenue;
+        uint256 platformRevenue;
+
+        RevenueShare[] memory revenueShares = saleData_.revenueShares;
+        uint256 remainingRev = itemRevenue;
+
+        // deduct advances payment from revenue
+        for (uint256 j = 0; j < revenueShares.length && remainingRev > 0; j++) {
+            uint256 remainingAdvanceAmount = advances[
+                revenueShares[j].recipient
+            ];
+            uint256 rev = remainingAdvanceAmount >= remainingRev
+                ? remainingRev
+                : remainingAdvanceAmount;
+            uint256 totalRev = saleData_.quantity * rev;
+            platformRevenue += totalRev;
+            advances[revenueShares[j].recipient] -= totalRev;
+            remainingRev -= totalRev;
+        }
+
+        // distribute revenue
+        if (remainingRev > 0) {
+            for (uint256 j = 0; j < revenueShares.length; j++) {
+                address recipient = revenueShares[j].recipient;
+                uint256 rev = (remainingRev * revenueShares[j].bps) / 10000;
+                uint256 totalRev = saleData_.quantity * rev;
+                if (recipient == costReceiver) {
+                    platformRevenue += totalRev;
+                    continue;
+                }
+                distributedRevenue += totalRev;
+                payable(recipient).transfer(totalRev);
```

Comment:
Should accumulate into one tx for the same recipient.

Reply:
we only use 1 revenue shares array for all the tokens, so we don't need to `accumulate` anymore